WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -1214,6 +1214,8 @@ private extension EditableOrderViewModel {
 
         return ProductInOrderViewModel(productRowViewModel: rowViewModel,
                                        productDiscountConfiguration: addProductDiscountConfiguration(on: orderItem),
+                                       showCouponsAndDiscountsAlert: orderSynchronizer.order.coupons.isNotEmpty &&
+                                                                     featureFlagService.isFeatureFlagEnabled(.ordersWithCouponsM4),
```

Comment:
This is a personal preference regarding flag visibility, so feel free to ignore it, but we could extract this to a class property:

```
private var shouldshowCouponsAndDiscountsAlert: Bool {
  orderSynchronizer.order.coupons.isNotEmpty &&
  featureFlagService.isFeatureFlagEnabled(.ordersWithCouponsM4)
}
``` 

```suggestion
showCouponsAndDiscountsAlert: shouldshowCouponsAndDiscountsAlert,
```